consensus/src/common.rs

```diff
@@ -4,12 +4,16 @@
 //! Implements common APIs for the blockchain in the context of consensus.
```

Comment:
This adds a bunch more of the difficult-to-read, repeated code tests we are trying to remove in #196. I don't think this is the way we should be writing these. Maybe chat with @creativcoder to see how they are planning to rewrite them and we can do it that way in the first place

Reply:
Yes, we need not manually populate test payloads this way. For the same reason, I wanted to have our base internal types i.e., `ProtocolState` annotated with `Serialize`, `Deserialize`? @willemolding is it possible for us to add the Serialize annotation to those; would they break anything?

Reply:
@willemolding  Update: We decided to remove this test and add it later as part of https://github.com/ChainSafe/mina-rs/issues/196 Additionally for removing the tech debt we'll need to have json serialization support from base types, which can be added after https://github.com/ChainSafe/mina-rs/pull/209 is merged. 
So merging this now.